### ISSUE TITLE
Pushing all logs in sentry irrespective of their level ? [want to push only error logs]

### ISSUE THREAD

**Issue by p00j4** [getsentry/sentry-java#109]
I'm having webdriver+Maven test automation project with logback implemented.
Had added raven-logback in the existing logging mechanism for sentry with 

```
<root level="error"><appender-ref ref="Sentry"/> </root>
```

however its pushing all of the logs instead of only Error Level 
may be because of existing mechanism  [check complete file down]

```
<root level="trace"><appender-ref ref="STDOUT" /><appender-ref ref="FILE" /><appender-ref ref="SIFT" /></root> 
```

What setting change has to be done for exclusively saying that 'only push logs of level "Error" or above ? (I want to push only logs which are errors)

Here are the configuration files [check screenshot also, I don't know why editor is replacing some tags with nothing ]

logback-tray.xml

```
<configuration debug="false" scan="false">

    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <pattern>[%5p] [%t] %m [%M::%F:%L] %n</pattern>
        </encoder>
    </appender>

    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
        <file>run.log</file>
        <encoder>
            <pattern>%d{ISO8601} [%5p] [%t] %m [%M::%F:%L] %n</pattern>
        </encoder>
    </appender>

    <appender name="SIFT" class="ch.qos.logback.classic.sift.SiftingAppender">

        <discriminator>
            <key>tcID</key>
            <defaultValue>FW</defaultValue>
        </discriminator>
        <sift>
            <appender name="FILE-${tcID}" class="ch.qos.logback.core.FileAppender">
                <filter class="com.project.qa.fw.util.TRAYLogEventFilter" />
                <file>SiftLogs/${tcID}.log</file>
                <append>false</append>
                <layout class="ch.qos.logback.classic.PatternLayout">
                    <pattern>%d{ISO8601} [%5p] [%t] %m [%M::%F:%L] %n</pattern>
                </layout>
            </appender>
        </sift>
    </appender>
  lt    root level="trace"   gt
        lt appender-ref ref="STDOUT"  gt
        lt appender-ref ref="FILE"  gt
     lt appender-ref ref="SIFT"  gt
    close root

    <!-- for sentry -->
     <appender name="Sentry" class="net.kencochrane.raven.logback.SentryAppender">
        <dsn>
            http://user:password@sentry.project.com/2
        </dsn>
    </appender>

    <root level="error">
        <appender-ref ref="Sentry"/>
    </root>
</configuration>
```

pom.xml

```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>ProjectGenFw</groupId>
    <artifactId>ProjectGenFw</artifactId>
    <version>0.0.1-SNAPSHOT</version>
    <name>Generic Fw</name>
    <properties>
        <logback.configurationFile>src/main/resources/logback-tray.xml</logback.configurationFile>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

    <dependencies>
        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>2.42.2</version>
            <!-- <version>2.24.1</version> -->
        </dependency>
<dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-classic</artifactId>
            <version>1.0.7</version>
        </dependency>
        <dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-core</artifactId>
            <version>1.0.7</version>
        </dependency>

        <dependency> <!-- pooja for sentry -->
            <groupId>net.kencochrane.raven</groupId>
            <artifactId>raven-logback</artifactId>
            <version>5.0</version>
        </dependency>

    </dependencies>
</project>
```

![screen shot 2014-08-21 at 7 45 27 pm](https://cloud.githubusercontent.com/assets/6470509/4007301/f139f356-29c1-11e4-8e1b-f767614a8547.png)

**Comment by weaselmetal** [comment]
Just ran into the same problem. ThresholdFilter seems to have no effect. All events end up in sentry.

**Comment by bretthoerner** [comment]
Can you post the logging config that didn't work?

**Comment by bretthoerner** [comment]
I'd like to see the full configuration that didn't work, because I suspect it's related to the documentation fix coming in: https://github.com/getsentry/raven-java/pull/306

A lot of our examples used to set multiple root loggers. I just tested the following config and it sent `DEBUG` to Console and only `WARN` and greater to Sentry:

```xml
<configuration>

  <appender name="Console" class="ch.qos.logback.core.ConsoleAppender">
    <encoder>
      <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
    </encoder>
  </appender>

  <appender name="Sentry" class="com.getsentry.raven.logback.SentryAppender">
    <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
      <level>WARN</level>
    </filter>
  </appender>

  <root level="DEBUG">
    <appender-ref ref="Console" />
    <appender-ref ref="Sentry"/>
  </root>

</configuration>
```

**Comment by weaselmetal** [comment]
Well, I'd swear that this config looks just like what I had, but for me the ThresholdFilter had no effect. INFO output ended up in Sentry even though the ThresholdFilter level was set to WARN.

Will check again tomorrow where I have the setup that made me reply today.

 

Gesendet: Dienstag, 07. Februar 2017 um 19:46 Uhr
Von: "Brett Hoerner" <notifications@github.com>
An: getsentry/raven-java <raven-java@noreply.github.com>
Cc: weaselmetal <stefan_w@gmx.at>, Mention <mention@noreply.github.com>
Betreff: Re: [getsentry/raven-java] Pushing all logs in sentry irrespective of their level ? [want to push only error logs] (#109)


I'd like to see the full configuration that didn't work, because I suspect it's related to the documentation fix coming in: #306

A lot of our examples used to set multiple root loggers. I just tested the following config and it sent DEBUG to Console and only WARN and greater to Sentry:


<configuration>

  <appender name="Console" class="ch.qos.logback.core.ConsoleAppender">
    <encoder>
      <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
    </encoder>
  </appender>

  <appender name="Sentry" class="com.getsentry.raven.logback.SentryAppender">
    <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
      <level>WARN</level>
    </filter>
  </appender>

  <root level="DEBUG">
    <appender-ref ref="Console" />
    <appender-ref ref="Sentry"/>
  </root>

</configuration>
